### ISSUE TITLE
Add Python3 support

### ISSUE THREAD

**Issue by mertyildiran** [DragonComputer/Dragonfire#15]
Add Python3 support project-wide

- [ ] Double-check everything works before sending the pull request
- [ ] Make sure you did not brake the Python2.7 support
- [ ] Check if Universal Wheels working correctly: `python setup.py bdist_wheel --universal`

**Comment by Tardog** [comment]
Is there already someone working on Python 3 support? If not: I would like to take a crack on this.

**Comment by mertyildiran** [comment]
@Tardog no one is working on this issue so sure you can take it. :blush:

**Comment by cclauss** [comment]
@Tardog I would recommend steps 0 and 1 of http://python-future.org/futurize_cheatsheet.html  Then see if step 2 is needed or not.  This is a safe approach that avoids a lot of tedious changes.

**Comment by Tardog** [comment]
@cclauss Thank you. I won't have the time to work on this before tomorrow, so feel free to add further contributions if you are able and willing to.

**Comment by cclauss** [comment]
https://travis-ci.org/DragonComputer/Dragonfire/jobs/290522020#L468 shows some of the issues that need to be addressed.

**Comment by Tardog** [comment]
Sadly, no progress so far, because I'm having trouble getting Dragonfire to work. There seems to be a problem with the decoder attempting to create a pipeline for one of the Gstreamer elements (onlinegmmdecodefaster). If I don't find a solution today, I'm going to create a new issue for that.

**Comment by mertyildiran** [comment]
@Tardog could you open a new issue and post the error there with full traceback?

**Comment by Tardog** [comment]
Here you go: #47

**Comment by mertyildiran** [comment]
Added with https://github.com/DragonComputer/Dragonfire/commit/cfa6bea2b1e14d10c6a9ddb152020ae1b29052e3

**Comment by cclauss** [comment]
Hi @Tardog ,  I apologize that I forgot about your claim above on this issue when I submitted my PR.  I hope that my submission did not offend.  The code in this repo is now compatible with both Python 2 and Python 3 and Travis CI + flake8 will let us know if we break that compatibility.  However, code that passes a linter is not necessarily code that works as expected in the real world. We have lot of external dependencies that might have their own Python 3 issues.  Therefor it would be wonderful if you could stress test Dragonfire running on Python 3 because I would bet dollars to doughnuts that there are more issues to find and fix before we can declare final victory on this one.

**Comment by Tardog** [comment]
Hi @cclauss – no apology necessary! Dragonfire is still resisting my best efforts to get it up and running, so I doubt I would have been able to work on this issue in due time. Thank you for taking this up. I’m definitely going to test with Python 3 when I finally find a way to run it.
